Dockerfile

```diff
@@ -17,10 +17,12 @@ FROM golang:1.14 as builder
 
 WORKDIR /sigs.k8s.io/external-dns
 
+COPY go.mod .
+COPY go.sum .
+RUN go mod download
```

Comment:
I think we don't explicitly need to run `go mod download` as `make build` will run `go build` which will download the needed dependencies with the version of Go that we use.

Reply:
It's an optimization in the docker caching mechanism.
If my go mod and go sum don't change why re-pull the deps when testing locally